src/hooks/useNewLiquidityMiningCampaign.ts

```diff
@@ -59,18 +45,23 @@ export function useNewLiquidityMiningCampaign(
       return null
     const formattedStartTime = Math.floor(startTime.getTime() / 1000).toString()
     const formattedEndTime = Math.floor(endTime.getTime() / 1000).toString()
-    if (targetedPairOrToken instanceof Pair && lpTokenTotalSupply) {
-      const { address, symbol, name, decimals } = targetedPairOrToken.liquidityToken
-      const lpTokenNativeCurrencyPrice = getLpTokenPrice(
-        targetedPairOrToken,
-        nativeCurrency,
-        lpTokenTotalSupply.raw.toString(),
-        targetedPairReserveNativeCurrency.raw.toString()
-      )
+    const derivedToken = targetedPairOrToken instanceof Pair ? targetedPairOrToken.liquidityToken : targetedPairOrToken
```

Comment:
`isPair` should be defined as we are checking this more than once

Reply:
I will make an issue to address all of this cases since Im not sure how to dry up that part...